plugins/CorePluginsAdmin/vue/src/PluginSettings/PluginSettings.vue

```diff
@@ -0,0 +1,260 @@
+
+      settingsPerPlugin.forEach((settings) => {
+        settings.settings.forEach((setting) => {
+          this.settingValues[`${settings.pluginName}.${setting.name}`] = setting.value;
+        });
+      });
+
+      window.anchorLinkFix.scrollToAnchorInUrl();
+    }).catch(() => {
+      this.isLoading = false;
+    });
+  },
+  computed: {
+    apiMethod(): string {
+      return this.mode === 'admin'
+        ? 'CorePluginsAdmin.getSystemSettings'
+        : 'CorePluginsAdmin.getUserSettings';
+    },
+    saveApiMethod(): string {
+      return this.mode === 'admin'
+        ? 'CorePluginsAdmin.setSystemSettings'
+        : 'CorePluginsAdmin.setUserSettings';
+    },
+  },
+  methods: {
+    save(requestedPlugin: string) {
+      const { saveApiMethod } = this;
+      const { root } = this.$refs;
+
+      const $root = $(root);
+
+      if (this.mode === 'admin' && !this.passwordConfirmation) {
+        this.settingsToSave = requestedPlugin;
+
+        const onEnter = (event) => {
+          const keycode = event.keyCode ? event.keyCode : event.which;
+          if (keycode === '13') {
+            $root.find('.confirm-password-modal').modal('close');
+            this.save(requestedPlugin);
+          }
+        };
+
+        $root.find('.confirm-password-modal').modal({
+          dismissible: false,
+          onOpenEnd: () => {
+            $('.modal.open #currentUserPassword').focus();
+            $('.modal.open #currentUserPassword').off('keypress').keypress(onEnter);
+          },
+        }).modal('open');
+
+        return;
+      }
+
+      this.isSaving[requestedPlugin] = true;
+
+      const settingValuesPayload = this.getValuesForPlugin(requestedPlugin);
+
+      AjaxHelper.post(
+        { method: saveApiMethod },
+        { settingValues: settingValuesPayload, passwordConfirmation: this.passwordConfirmation },
+      ).then(() => {
+        this.isSaving[requestedPlugin] = false;
+
+        NotificationsStore.show({
+          message: translate('CoreAdminHome_PluginSettingsSaveSuccess'),
+          id: 'generalSettings',
+          context: 'success',
+          type: 'transient',
+        });
+        NotificationsStore.scrollToNotification('generalSettings');
+      }).catch(() => {
+        this.isSaving[requestedPlugin] = false;
+      });
+
+      this.passwordConfirmation = '';
+      this.settingsToSave = null;
+    },
+    makeSettingConditionFunction(setting: Setting, pluginName: string) {
+      const { condition } = setting;
+      if (!condition) {
+        return undefined;
+      }
+
+      return () => {
+        if (!conditionScope) {
+          const $rootScope = Matomo.helper.getAngularDependency('$rootScope');
+          conditionScope = $rootScope.$new(true);
+        }
+
+        // TODO: this is definitely not as performant. would probably need a separate component
+        // for a single plugin's settings so we can make this and other types of transforms
+        // computed properties.
```

Comment:
If that's useful and/or needed, might be good to create an follow up issue on github instead of a code comment. Otherwise I guess this will be never addressed 🙈